### PR TITLE
chore: merge release/v0.3.1 back to main

### PR DESCRIPTION
**Pull request by KayMKM**
## Summary

Merge the published `release/v0.3.1` tip (`ded1fc35ca95ab07dea51c9e817263aafcd9520d`) back into `main`.

- Bring back the v0.3.1 version bump and CHANGELOG entry (#1393).
- Carry the QNN Catalog acquisition and wheel-packaging fixes without installing a conflicting standard ONNX Runtime distribution (#1395).
- Carry the user-writable VitisAI cache directory, HF export/mask fixes, decoder invocation fixes, and release-pipeline uv update (#1378, #1379, #1384, #1374).

## Merge conflict resolution

- Preserve `main`'s `onnxruntime-windowsml>=1.24.5,<1.28` range for OpenVINO operator profiling. Make the wheel regression test validate the installed runtime against the wheel's dependency metadata rather than hardcoding the 1.24 series.
- Keep the removed Transformers compatibility shim removed. Port the export-scoped integral SDPA mask handling to the existing `export/_attention.py` helper, with coverage for positional/keyword masks and restoration after failure.
- Preserve compile-cache file identity tracking and the existing `serve` CI coverage while adding packaging coverage.
- Leave `main`'s GitHub App service connection and UTF-8 release-note fixes intact (#1292, #1293).

The published v0.3.1 tag, release branch, and release assets are unchanged.

### CODE REVIEW

**Comment by xieofxie** [comment]
need to fix test

**Review by DingmaomaoBJTU — APPROVED** [review]

File: tests/unit/export/test_htp_exporter_attention_compat.py [hunk 1]
```diff
@@ -320,3 +319,43 @@
...
+    model = _NestedAttentionModel()
+    original_sdpa = torch.nn.functional.scaled_dot_product_attention
+
+    with (
+        pytest.raises(RuntimeError, match="export failed"),
+        HTPExporter()._export_compatibility_context(model, _export_config(eager_attention=True)),
+    ):
+        raise RuntimeError("export failed")
+
+    assert torch.nn.functional.scaled_dot_product_attention is original_sdpa
```
[the remark this answers is not in the document]
Comment (KayMKM) [reply tests/unit/export/test_htp_exporter_attention_compat.py:361]: Updated in 555c989c: use an explicit outer `pytest.raises` block and inner export context so exception suppression is visible to control-flow analysis. The post-exception restoration assertions are retained. The affected pytest scope passes (100 tests), as do Ruff and mypy; the new CodeQL run will verify the alert.

File: tests/unit/export/test_htp_exporter_attention_compat.py [hunk 2]
```diff
@@ -320,3 +319,43 @@
...
+        assert model.config._attn_implementation == "eager"
+
+    torch.testing.assert_close(actual, expected)
+    assert torch.nn.functional.scaled_dot_product_attention is original_sdpa
+    assert model.config._attn_implementation == "sdpa"
+
+
+def test_export_context_restores_sdpa_after_failure() -> None:
+    model = _NestedAttentionModel()
+    original_sdpa = torch.nn.functional.scaled_dot_product_attention
```
[the remark this answers is not in the document]
Comment (KayMKM) [reply tests/unit/export/test_htp_exporter_attention_compat.py:352]: Addressed in 555c989c with the same explicit nesting change. `original_sdpa` remains necessary: the assertion after the exception checks that the original SDPA callable is restored. That assertion and the eager-attention restoration test pass locally; CodeQL verification is pending on the pushed commit.

**Review by DingmaomaoBJTU — APPROVED** [review]
